spec/setter_methods_spec.rb

```diff
@@ -25,24 +27,53 @@
 
     before { allow(Time).to receive(:now).and_return(now) }
 
-    it "immediately dispatches a validation-free update" do
+    it "performs an update on the relevant attribute" do
       register.active!
       register.close!
 
+      register.reload
+
       expect(register).to be_active
       expect(register).to be_closed
     end
 
     it "updates default timestamp columns with the current date and time" do
       register.fill!
+
+      register.reload
+
       expect(register).to be_full
       expect(register.full_at).to eq(now)
     end
 
     it "updates custom timestamp columns with the current date and time" do
       register.empty!
+
+      register.reload
+
       expect(register).to be_empty
       expect(register.emptied_at).to eq(now)
     end
+
+    context "when there are validation errors" do
+      before { register.include_validations = true }
+
+      specify { expect(register).to_not be_valid }
+
+      it "raises and does not persist any changes" do
+        expect { register.empty! }.to raise_error(ActiveRecord::RecordInvalid, "Validation failed: Required attribute can't be blank")
+        expect { register.close! }.to raise_error(ActiveRecord::RecordInvalid, "Validation failed: Required attribute can't be blank")
+
+        expect(register).to be_empty
+        expect(register).to be_closed
+        # TODO: is this actually expected if the values weren't persisted?
```

Comment:
I thought this might be worth discussing.

In the case where `empty!` actually fails to persist, is it expected that `register.emptied_at = Time.now` sticks around on the in-memory record? My own instinct is that yes, it's expected, as `empty!` is basically a convenience method around handling setter timestamps.